packages/imperative/src/rest/src/client/AbstractRestClient.ts

```diff
@@ -229,6 +229,27 @@ export abstract class AbstractRestClient {
         ImperativeExpect.toNotBeNullOrUndefined(mSession);
         this.mLogger = Logger.getImperativeLogger();
         this.mIsJson = false;
+
+        /* Set the order of precedence in which available credentials will be used.
+         *
+         * The Zowe SDK policy is to select password credentials over a token.
+         * However, this class was originally released with the token over password.
+         * The commonly-used ConnectionPropsForSessCfg.resolveSessCfgProps enforces the
+         * order of password over token. None-the-less, consumers which directly extended
+         * AbstractRestClient came to rely on the order of token over password.
+         *
+         * Later changes in this class to adhere to Zowe policy inadvertently broke
+         * such extenders. While we now use a generalized authTypeOrder property to
+         * determine the order, until a means is provided for consumers (and/or end-users)
+         * to customize their credential order of precedence, we hard-code the
+         * original order of token over password to correct the breaking change.
+         */
+        this.mSession.ISession.authTypeOrder = [
+            SessConstants.AUTH_TYPE_TOKEN,
+            SessConstants.AUTH_TYPE_BASIC,
```

Comment:
Do we want to reverse the order here (basic > Token) since this is against the `next` branch (to be consistent)?
or keep it as is?

Reply:
Even though we are "allowed" to make breaking changes in V3, I think that the same extenders will hit the same dead-end failure with no practical solution. Their products will be broken.

I think that we will have to keep the current credential order in place until we implement a means for extenders (and maybe end-users) to specify their preferred order of credentials. I don't think that we know when we will undertake such an enhancement. I also cannot predict if that enhancement can be done in a non-breaking fashion after V3 is released. None-the-less, I think we have to stick with the current order for now.